`jx boot` doesn't like I would not give a feedback

### Summary 

```
? jx boot has only be validated on GKE, we'd love feedback and contributions for other Kubernetes providers No

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x100 pc=0x4641787]

goroutine 1 [running]:
github.com/jenkins-x/jx/pkg/cmd/step/verify.(*StepVerifyPreInstallOptions).ValidateRequirements(0xc000688f00, 0x0, 0xc0005d9040, 0x40, 0x0, 0x0)
        /workspace/source/pkg/cmd/step/verify/step_verify_preinstall.go:704 +0x37
github.com/jenkins-x/jx/pkg/cmd/step/verify.(*StepVerifyPreInstallOptions).Run(0xc000688f00, 0x0, 0x0)
        /workspace/source/pkg/cmd/step/verify/step_verify_preinstall.go:92 +0xf5
github.com/jenkins-x/jx/pkg/cmd/step/verify.NewCmdStepVerifyPreInstall.func1(0xc000627180, 0xaa30de0, 0x0, 0x0)
        /workspace/source/pkg/cmd/step/verify/step_verify_preinstall.go:66 +0x6f
github.com/spf13/cobra.(*Command).execute(0xc000627180, 0xaa30de0, 0x0, 0x0, 0xc000627180, 0xaa30de0)
        /workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:766 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000b8f00, 0xc0002c1b40, 0x62178e0, 0xc0000c4000)
        /workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2ec
github.com/spf13/cobra.(*Command).Execute(...)
        /workspace/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800
github.com/jenkins-x/jx/cmd/jx/app.Run(0x0, 0x0, 0x0, 0xc000713f88, 0xc0000ae058)
        /workspace/source/cmd/jx/app/jx.go:19 +0x185
main.main()
        /workspace/source/cmd/jx/jx.go:11 +0x32
error: failed to interpret pipeline file jenkins-x.yml: failed to run '/bin/sh -c jx step verify preinstall' command in directory 'env', output: ''
```

### Steps to reproduce the behavior

`jx boot`

### Expected behavior

Respect to my choice? :-)

### Actual behavior

Crash

### Jx version

The output of `jx version` is:

```
jx                 2.0.689
Kubernetes cluster v1.13.10-eks-5ac0f1
kubectl            v1.15.3
git                2.23.0
Operating System   Mac OS X 10.14.6 build 18G95
```

### Jenkins type

<!--
Select which installation type are you using. 
-->
- [x] Serverless Jenkins X Pipelines (Tekton + Prow)
- [ ] Classic Jenkins

### Kubernetes cluster

The cluster created using `eksctl`

### Operating system / Environment

Mac OS X 10.14.6 build 18G95

--- a/pkg/cloud/amazon/common.go
+++ b/pkg/cloud/amazon/common.go
@@ -4,8 +4,10 @@ import (
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/pkg/errors"
 	"os"
 	"path"
+	"regexp"
 	"runtime"
 )
 
@@ -59,6 +61,16 @@ func ResolveRegion(profileOption string, regionOption string) (string, error) {
 
 func ResolveRegionWithoutOptions() (string, error) {
 	return ResolveRegion("", "")
+}
+
+// ParseContext parses the EKS cluster context to extract the cluster name and the region
+func ParseContext(context string) (string, string, error) {
+	reg := regexp.MustCompile(`([a-zA-Z][-a-zA-Z0-9]*)\.((us(-gov)?|ap|ca|cn|eu|sa)-(central|(north|south)?(east|west)?)-\d)\.*`)
+	result := reg.FindStringSubmatch(context)
+	if len(result) < 3 {
+		return "", "", errors.Errorf("unable to parse %s as <cluster_name>.<region>.*", context)
+	}
+	return result[1], result[2], nil
 }
 
 // UserHomeDir returns the home directory for the user the process is running under.

--- a/pkg/cloud/amazon/common_test.go
+++ b/pkg/cloud/amazon/common_test.go
@@ -110,3 +110,37 @@ func TestReadingRegionFromEnvProfile(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, "qux", *session.Config.Region)
 }
+
+func TestParseContext(t *testing.T) {
+	t.Parallel()
+	tests := map[string]struct {
+		context string
+		cluster string
+		region  string
+	}{
+		"full cluster name from eksctl": {
+			context: "cluster-name-jx.us-east-1.eksctl.io",
+			cluster: "cluster-name-jx",
+			region:  "us-east-1",
+		},
+		"full cluster name no eksctl": {
+			context: "cluster-name-jx.us-east-1",
+			cluster: "cluster-name-jx",
+			region:  "us-east-1",
+		},
+		"full cluster name other region": {
+			context: "cluster-name-jx.eu-north-4",
+			cluster: "cluster-name-jx",
+			region:  "eu-north-4",
+		},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			cluster, region, err := amazon.ParseContext(tc.context)
+			assert.NoErrorf(t, err, "there shouldn't be an error parsing context %s", tc.context)
+			assert.Equal(t, tc.cluster, cluster)
+			assert.Equal(t, tc.region, region)
+		})
+	}
+}

--- a/pkg/cmd/step/verify/step_verify_preinstall.go
+++ b/pkg/cmd/step/verify/step_verify_preinstall.go
@@ -2,6 +2,7 @@ package verify
 
 import (
 	"fmt"
+	"github.com/jenkins-x/jx/pkg/cloud/amazon"
 	"io/ioutil"
 	"os"
 	"strings"
@@ -419,6 +420,7 @@ func (o *StepVerifyPreInstallOptions) VerifyInstallConfig(kubeClient kubernetes.
 			modifyMapIfNotBlank(configMap.Data, kube.ProjectID, requirements.Cluster.ProjectID)
 			modifyMapIfNotBlank(configMap.Data, kube.ClusterName, requirements.Cluster.ClusterName)
 			modifyMapIfNotBlank(configMap.Data, secrets.SecretsLocationKey, secretsLocation)
+			modifyMapIfNotBlank(configMap.Data, kube.Region, requirements.Cluster.Region)
 			return nil
 		}, nil)
 	if err != nil {
@@ -435,11 +437,18 @@ func (o *StepVerifyPreInstallOptions) gatherRequirements(requirements *config.Re
 		if requirements.Cluster.Provider == "" {
 			return nil, errors.Errorf(msg, "provider")
 		}
-		if requirements.Cluster.ProjectID == "" {
-			return nil, errors.Errorf(msg, "project")
+		if requirements.Cluster.Provider == cloud.EKS || requirements.Cluster.Provider == cloud.AWS {
+			if requirements.Cluster.Region == "" {
+				return nil, errors.Errorf(msg, "region")
+			}
 		}
-		if requirements.Cluster.Zone == "" {
-			return nil, errors.Errorf(msg, "zone")
+		if requirements.Cluster.Provider == cloud.GKE {
+			if requirements.Cluster.ProjectID == "" {
+				return nil, errors.Errorf(msg, "project")
+			}
+			if requirements.Cluster.Zone == "" {
+				return nil, errors.Errorf(msg, "zone")
+			}
 		}
 		if requirements.Cluster.EnvironmentGitOwner == "" {
 			return nil, errors.Errorf(msg, "environmentGitOwner")
@@ -453,6 +462,13 @@ func (o *StepVerifyPreInstallOptions) gatherRequirements(requirements *config.Re
 		requirements.Cluster.Provider, err = util.PickName(cloud.KubernetesProviders, "Select Kubernetes provider", "the type of Kubernetes installation", o.In, o.Out, o.Err)
 		if err != nil {
 			return nil, errors.Wrap(err, "selecting Kubernetes provider")
+		}
+	}
+
+	if requirements.Cluster.Provider != cloud.GKE {
+		// lets check we want to try installation as we've only tested on GKE at the moment
+		if !o.showProvideFeedbackMessage() {
+			return requirements, errors.New("finishing execution")
 		}
 	}
 
@@ -522,18 +538,47 @@ func (o *StepVerifyPreInstallOptions) gatherRequirements(requirements *config.Re
 				log.Logger().Info("no need to reconnect to cluster")
 			}
 		}
-	} else {
-		// lets check we want to try installation as we've only tested on GKE at the moment
-		confirmed := util.Confirm("jx boot has only be validated on GKE, we'd love feedback and contributions for other Kubernetes providers",
-			true, "", o.In, o.Out, o.Err)
-		if !confirmed {
-			return nil, nil
+	} else if requirements.Cluster.Provider == cloud.EKS || requirements.Cluster.Provider == cloud.AWS {
+		var currentRegion, currentClusterName string
+		var autoAcceptDefaults bool
+		if requirements.Cluster.Region == "" || requirements.Cluster.ClusterName == "" {
+			kubeConfig, _, err := o.Kube().LoadConfig()
+			if err != nil {
+				return nil, errors.Wrapf(err, "loading kubeconfig")
+			}
+			context := kube.Cluster(kubeConfig)
+			currentClusterName, currentRegion, err = amazon.ParseContext(context)
+			if currentClusterName != "" && currentRegion != "" {
+				log.Logger().Infof("")
+				log.Logger().Infof("Currently connected cluster is %s in region %s", util.ColorInfo(currentClusterName), util.ColorInfo(currentRegion))
+				autoAcceptDefaults = util.Confirm(fmt.Sprintf("Do you want to jx boot the %s cluster?", util.ColorInfo(currentClusterName)), true, "Enter Y to use the currently connected cluster or enter N to specify a different cluster", o.In, o.Out, o.Err)
+			} else {
+				log.Logger().Infof("Enter the cluster you want to jx boot")
+			}
+		}
+
+		if requirements.Cluster.Region == "" {
+			if autoAcceptDefaults && currentRegion != "" {
+				requirements.Cluster.Region = currentRegion
+			}
+		}
+		if requirements.Cluster.ClusterName == "" {
+			if autoAcceptDefaults && currentClusterName != "" {
+				requirements.Cluster.ClusterName = currentClusterName
+			} else {
+				requirements.Cluster.ClusterName, err = util.PickValue("Cluster name", currentClusterName, true,
+					"The name for your cluster", o.In, o.Out, o.Err)
+				if err != nil {
+					return nil, errors.Wrap(err, "getting cluster name")
+				}
+			}
 		}
 	}
 
 	requirements.Cluster.Provider = strings.TrimSpace(strings.ToLower(requirements.Cluster.Provider))
 	requirements.Cluster.ProjectID = strings.TrimSpace(requirements.Cluster.ProjectID)
 	requirements.Cluster.Zone = strings.TrimSpace(strings.ToLower(requirements.Cluster.Zone))
+	requirements.Cluster.Region = strings.TrimSpace(strings.ToLower(requirements.Cluster.Region))
 	requirements.Cluster.ClusterName = strings.TrimSpace(strings.ToLower(requirements.Cluster.ClusterName))
 
 	err = o.gatherGitRequirements(requirements)
@@ -829,4 +874,14 @@ func modifyMapIfNotBlank(m map[string]string, key string, value string) {
 	} else {
 		log.Logger().Debugf("Cannot update key %s, map is nil", key)
 	}
+}
+
+func (o *StepVerifyPreInstallOptions) showProvideFeedbackMessage() bool {
+	log.Logger().Info("jx boot has only been validated on GKE, we'd love feedback and contributions for other Kubernetes providers")
+	if !o.BatchMode {
+		return util.Confirm("Continue execution anyway?",
+			true, "", o.In, o.Out, o.Err)
+	}
+	log.Logger().Info("Running in Batch Mode, execution will continue")
+	return true
 }

